### PR TITLE
Check for ShadowComponent in UVOL

### PR DESCRIPTION
**Pull request by CITIZENDOT**
## References
closes #_insert number here_

### CODE REVIEW

File: packages/engine/src/scene/components/UVOL1Component.ts [hunk 1]
```diff
@@ -160,9 +161,11 @@ function UVOL1Reactor() {
     if (volumetric.useLoadingEffect.value) {
       setComponent(entity, UVOLDissolveComponent)
     }
-    const shadow = getMutableComponent(entity, ShadowComponent)
-    shadow.cast.set(true)
-    shadow.receive.set(true)
+    const shadow = getOptionalMutableComponent(entity, ShadowComponent)
```
Comment (HexaField) [inline packages/engine/src/scene/components/UVOL1Component.ts:164]: we should put these in other useEffects
```ts
const shadow = useOptionalComponent(entity, ShadowComponent)
useEffect(() => {
    if (shadow) {
      shadow.cast.set(true)
      shadow.receive.set(true)
    }
    }, [shadow])
```

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/scene/components/UVOL1Component.ts (hunk 1 above)
Comment (CITIZENDOT) [reply packages/engine/src/scene/components/UVOL1Component.ts]: Done

File: packages/engine/src/scene/components/UVOL2Component.ts [hunk 2]
```diff
@@ -460,6 +452,18 @@ transformed.z += mix(keyframeA.z, keyframeB.z, mixRatio);
     }
   }, [])
 
+  useEffect(() => {
+    if (!shadow) return
+    if (manifest.current.type === UVOL_TYPE.UNIFORM_SOLVE_WITH_COMPRESSED_TEXTURE && shadow) {
```
Comment (HexaField) [inline packages/engine/src/scene/components/UVOL2Component.ts:457]: these second `shadow` checks are unnecessary

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/scene/components/UVOL2Component.ts (hunk 2 above)
Comment (CITIZENDOT) [reply packages/engine/src/scene/components/UVOL2Component.ts]: Fixed.

**Review by HexaField — APPROVED** [review]

**Pull request merged**